src/mux_client_impl/command.rs

```diff
@@ -0,0 +1,103 @@
+use super::Error;
+use super::RemoteChild;
+use super::{as_raw_fd, ChildStderr, ChildStdin, ChildStdout, Stdio};
+
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+use openssh_mux_client::connection::{Connection, EstablishedSession, Session};
+
+#[derive(Debug)]
+pub struct Command {
+    cmd: String,
+    ctl: PathBuf,
+
+    stdin_v: Stdio,
+    stdout_v: Stdio,
+    stderr_v: Stdio,
+}
+
+impl Command {
+    pub(crate) fn new(ctl: PathBuf, cmd: String) -> Self {
+        Self {
+            cmd,
+            ctl,
+
+            stdin_v: Stdio::null(),
+            stdout_v: Stdio::null(),
+            stderr_v: Stdio::null(),
+        }
+    }
+}
+
+impl Command {
+    pub fn arg<S: AsRef<str>>(&mut self, arg: S) -> &mut Self {
+        self.cmd.push_str(" '");
+        self.cmd.push_str(arg.as_ref());
+        self.cmd.push('\'');
+        self
+    }
+
+    pub fn raw_arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        self.cmd.push(' ');
+        self.cmd.push_str(&arg.as_ref().to_string_lossy());
+        self
+    }
```

Comment:
Unfortunately, the [ssh multiplex protocol](https://github.com/openssh/openssh-portable/blob/e665ed2d0c24fe11d5470ce72fa1e187377d3fc4/PROTOCOL.mux#L77) does not support passing arguments separately.

Honestly, that protocol is a mess.

Reply:
Ooof, that's very awkward indeed. I think at the very least then you're going to want to escape each `arg` here, rather than just blindly stick them in `'`, as otherwise users will get into painful problems if they send an argument that is, say, `"what's up"`.

Reply:
I tried the escaping here but then the integration test somehow failed.

Reply:
Then we need to dig into why that is. It's _definitely_ the case that if you pass the command line just as a single string, we need to do some kind of escaping here to have the remote end receive the same thing. Could it be that you end up double-escaping somehow?

Reply:
It could be, I will have a try.

Reply:
You are right.

Now I have removed `{native_mux_impl, process_impl}::Command::arg` and replaced them with `Command::arg` which simply calls `self.raw_arg` with escaped string.